src/core/AutomationClip.cpp

```diff
@@ -890,25 +890,20 @@ gui::ClipView * AutomationClip::createView( gui::TrackView * _tv )
 
 bool AutomationClip::isAutomated( const AutomatableModel * _m )
 {
-	TrackContainer::TrackList l;
-	l += Engine::getSong()->tracks();
-	l += Engine::patternStore()->tracks();
-	l += Engine::getSong()->globalAutomationTrack();
-
-	for( TrackContainer::TrackList::ConstIterator it = l.begin(); it != l.end(); ++it )
+	auto l = combineAllTracks();
+	for (const auto track : l)
 	{
-		if( ( *it )->type() == Track::AutomationTrack ||
-			( *it )->type() == Track::HiddenAutomationTrack )
+		if (track->type() == Track::AutomationTrack || track->type() == Track::HiddenAutomationTrack)
 		{
-			const Track::clipVector & v = ( *it )->getClips();
-			for( Track::clipVector::ConstIterator j = v.begin(); j != v.end(); ++j )
+			const Track::clipVector & clips = track->getClips();
+			for (const auto clip : clips)
 			{
-				const AutomationClip * a = dynamic_cast<const AutomationClip *>( *j );
-				if( a && a->hasAutomation() )
+				const AutomationClip * a = dynamic_cast<const AutomationClip*>(clip);
+				if (a && a->hasAutomation())
 				{
-					for( objectVector::const_iterator k = a->m_objects.begin(); k != a->m_objects.end(); ++k )
+					for (const auto obj : a->m_objects)
```

Comment:
GCC warns about copying `QPointer<AutomatableModel>`, but I think it doesn't hurt here.